src/Middleware/RequestThrottling/test/MiddlewareTests.cs

```diff
@@ -96,52 +71,39 @@ public async void RequestsBlockedIfQueueFull()
         }
 
         [Fact]
-        public async void FullQueueResultsIn503Error()
-        {
-            var middleware = TestUtils.CreateBlockingTestMiddleware(requestQueueLimit: 0);
-
-            var context = new DefaultHttpContext();
-            await middleware.Invoke(context).OrTimeout();
-            Assert.Equal(503, context.Response.StatusCode);
-        }
-
-        [Fact]
-        public void MultipleRequestsFillUpQueue()
+        public void IncomingRequestsFillUpQueue()
         {
             var middleware = TestUtils.CreateTestMiddleware(
-                maxConcurrentRequests: 1,
-                requestQueueLimit: 10,
-                next: httpContext =>
-                {
-                    return Task.Delay(TimeSpan.FromSeconds(30));
-                });
+                queue: TestStrategy.AlwaysBlock);
 
-            Assert.Equal(0, middleware.ActiveRequestCount);
+            Assert.Equal(0, middleware.TotalRequestCount);
 
-            var _ = middleware.Invoke(new DefaultHttpContext());
-            Assert.Equal(1, middleware.ActiveRequestCount);
+            _ = middleware.Invoke(new DefaultHttpContext());
+            Assert.Equal(1, middleware.TotalRequestCount);
 
             _ = middleware.Invoke(new DefaultHttpContext());
-            Assert.Equal(2, middleware.ActiveRequestCount);
+            Assert.Equal(2, middleware.TotalRequestCount);
         }
 
         [Fact]
-        public async void FullQueueInvokesOnRejected()
+        public async Task CleanupHappensEvenIfNextErrors()
```

Comment:
I'm liking the more generic names. This is the new version of SemaphoreStatePreservedIfRequestsError, right?

Reply:
yes